File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/service/ZeppelinServiceExceptions.scala
package org.intellij.plugin.zeppelin.service

import org.intellij.plugin.zeppelin.models.Interpreter

class InterpreterException(interpreter: Interpreter) extends Exception {
  override def getMessage: String = "Reason: " + interpreter.errorReason.getOrElse("unavailable")
}

class InterpreterNotFoundException(id: String) extends Exception {
  override def getMessage: String = s"An interpreter with id $id is not found"
}

class NotebookException() extends Exception

class ServiceIsUnavailableException() extends Exception {
  override def getMessage: String = "Service is unavailable, please restart the service"
}

class NoSelectedFilesException() extends Exception {
  override def getMessage: String = "Please, select the file for this operation"
}

File: zeppelin/implementation/src/org/intellij/plugin/zeppelin/service/ZeppelinServiceExceptions.kt
package org.intellij.plugin.zeppelin.service

import org.intellij.plugin.zeppelin.models.Interpreter
import org.intellij.plugin.zeppelin.models.ZeppelinException

class InterpreterException(private val interpreter: Interpreter) : ZeppelinException() {
    override val message: String
        get() = "Reason: " + (interpreter.errorReason ?: "unavailable")
}

class InterpreterNotFoundException(private val id: String) : ZeppelinException() {
    override val message: String
        get() = "An interpreter with id $id is not found"
}

class NoSelectedFilesException : ZeppelinException() {
    override val message: String
        get() = "Please, select the file for this operation"
}
